src/tokens/tokens.module.ts

```diff
@@ -31,7 +32,13 @@ import { BlockchainConnectorService } from './blockchain.service';
     EventStreamProxyModule,
   ],
   controllers: [TokensController],
-  providers: [TokensService, AbiMapperService, BlockchainConnectorService],
+  providers: [
+    ...MetricProviders,
+    LoggingAndMetricsInterceptor,
```

Comment:
Is it necessary to add these providers to individual modules if they've been registered globally?

Reply:
I found that without adding the providers to the tokens module I hit

```
[Nest] 16531  - 06/20/2023, 10:00:41 AM   ERROR [ExceptionHandler] Nest can't resolve dependencies of the BlockchainConnectorService (HttpService, ?). Please make sure that the argument LoggingAndMetricsInterceptor at index [1] is available in the TokensModule context.

Potential solutions:
- If LoggingAndMetricsInterceptor is a provider, is it part of the current TokensModule?
- If LoggingAndMetricsInterceptor is exported from a separate @Module, is that module imported within TokensModule?
  @Module({
    imports: [ /* the Module containing LoggingAndMetricsInterceptor */ ]
  })
```

I'm not sure if there's an alternative way which would mean individual modules don't need to add them?